lib/ueberauth/strategy/okta/oauth.ex

```diff
@@ -40,9 +40,10 @@ defmodule Ueberauth.Strategy.Okta.OAuth do
   Ueberauth.
   """
   def client(opts \\ []) do
+    config = Keyword.take(opts, [:client_id, :client_secret, :site])
+    config = if config != [], do: config, else: Application.fetch_env!(:ueberauth, __MODULE__)
```

Comment:
```suggestion
    config = Application.fetch_env!(:ueberauth, __MODULE__)
      @defaults
      |> Keyword.merge(config)
      |> Keyword.merge(opts)
      |> validate_config_option!(:client_id)
      |> validate_config_option!(:client_secret)
      |> validate_config_option!(:site)
      |> Client.new()
      |> OAuth2.Client.put_serializer("application/json", Jason)
```

Or something like that. That way we can reduce all the fetching and checking just a bit